### ISSUE TITLE
confused about posts_per_page

### ISSUE THREAD

**Issue by RRaphaell** [kevinzg/facebook-scraper#829]
I don't understand how post_per_page works. 
as I know it should call a callback function after every page
for example in this case, callback should be printed after 4 post

```
def handle_pagination_url(url):
    print("callback")

posts = get_posts(page, credentials=c,
                  options={"comments": True, 
                           "reactors": True, 
                           "posts_per_page": 4, 
                           "allow_extra_requests": False,
                           "sharers": True},
                  page_limit=None, extra_info=True, timeout=60,
                  start_url=start_url, request_url_callback=handle_pagination_url)

for i, post in enumerate(posts):
    print("saving post")
```
but I get this output. callback printed after 10 post
```
callback
saving post
saving post
saving post
saving post
saving post
saving post
saving post
saving post
saving post
saving post
callback
```

**Comment by neon-ninja** [comment]
posts_per_page only works for Pages, not Profiles

**Comment by RRaphaell** [comment]
> posts_per_page only works for Pages, not Profiles

yes, I run it for pages

**Comment by neon-ninja** [comment]
What Page are you having this problem with?

**Comment by RRaphaell** [comment]
for example https://www.facebook.com/PUMA

**Comment by neon-ninja** [comment]
I'm unable to reproduce this problem, I get this output for PUMA:

```
callback
callback
saving post
saving post
saving post
callback
saving post
saving post
saving post
saving post
callback
saving post
saving post
saving post
saving post
callback
saving post
saving post
saving post
saving post
callback
```

**Comment by RRaphaell** [comment]
can you check that page? https://www.facebook.com/europeangeorgia.ge

**Comment by neon-ninja** [comment]
yeah europeangeorgia.ge uses pagination URLs like https://m.facebook.com/profile/timeline/stream/?cursor=AQHRQpbmplnZYYzl4oihdV7Md5AZCKzUFu4viBygH57HtD3iCsLAarxlby0A8iiXon4ClZjHAb4scuIXTzKkiy8uJU4X-8sJGO1vhoAHyYv_sVqQVE4te0RTSPWQwgjWCT3eEM7mcghMzikfGFRFrU6cpr1kwyv6yoOfj9ZzOw1B1hsdg6V7CK3n80-7zAewj9l_&start_time=-9223372036854775808&profile_id=100044393380868&replace_id=u_0_0_lH\, there's no posts_per_page parameter in that URL, so posts_per_page won't work. Perhaps this Page was converted from a Profile or something. It has things that Profiles typically have .

**Comment by RRaphaell** [comment]
Then do yo have any ideas how to handle this? https://github.com/kevinzg/facebook-scraper/issues/824

If there are not post_per_page how can I switch accounts and continue scraping after last post

**Comment by neon-ninja** [comment]
You would have to keep a counter, and change accounts after some multiple of 4. So if you change accounts every 12 pages, that'd be every 48 posts

**Comment by GrimesSheenaLana** [comment]
> You would have to keep a counter, and change accounts after some higher number of pages. So if you change accounts every 12 pages, that'd be every 48 posts

@neon-ninja Sorry for interrupting you. I am wondering how to change accounts/cookies during the parsing. I am parsing posts and comments from one public Facebook page. It seems that no matter what my "**pages**" and "**posts per page**" values are, the parsing will stop around July 2021 (my intended date would be Nov 2019). Below is my codes, thanks so much!

```
from facebook_scraper import get_posts

cookie="facebook.com_cookies.txt"
df=pd.DataFrame()


for post in get_posts('sghealthministry',pages=1000,cookies=cookie,options={"comments": True,"post_per_page": 300}):
    df_temp = pd.DataFrame.from_dict(post, orient='index')
    df_temp = df_temp.T
    df=pd.concat([df,df_temp])
```
